### DISCUSSION TITLE
The mesh generating for cohesive zone model (CZM)

### DISCUSSION THREAD

**Discussion by RECHOA** [idaholab/moose#22149] [Q&A Modules: Solid mechanics] [answered]
Hi all,
I am new  to CZM and when i study the example case in MOOSE :([moose/modules/tensor_mechanics/test/tests/cohesive_zone_model/stretch_rotate_large_deformation.i](https://github.com/idaholab/moose/blob/bd7c088ec4/modules/tensor_mechanics/test/tests/cohesive_zone_model/stretch_rotate_large_deformation.i)),
 I noticed that the mesh and the  `interface` (boundary = 'interface') was generated by using `BreakMeshByBlockGenerator`.
![pic](https://user-images.githubusercontent.com/33534979/191304048-6b8a9197-9d50-4bc0-9bfd-3674e8a61ef5.jpg)
Now i want to generate the mesh in that case by using software Cubit. Whether just the `boundary = 'interface'` in inputfile need change to `boundary = 'surface1 surface2'` , 
or maybe i should merge  `surface1 ` and  `surface2` in Cubit and then set the surface as `interface` ?

**Comment by hugary1995** [comment, marked as answer]
An interface includes two opposite sides, e.g. a primary side and a secondary side. But the CZM implementation in tensor_mechanics operates on only one side of the interface. That was a design choice.

So in your case you should either provide `boundary = 'surface1'`, or `boundary = 'surface2'`, but not both. Also note that the mesh you supplied into MOOSE should be fully connected, i.e. the interface between surface1/surface2 is not disjoint. The splitting has to be handled by `BreakMeshByBlockGenerator`.

**Reply by RECHOA** [reply]
>  Also note that the mesh you supplied into MOOSE should be fully connected, i.e. the interface between surface1/surface2 is not disjoint. The splitting has to be handled by BreakMeshByBlockGenerator.

So i should merge  surface1 and surface2 to make volume1 and volume2 fully connected by using Cubit, and name them block1/block2, then use `BreakMeshByBlockGenerator` to split them?

**Reply by hugary1995** [reply]
Yes. The mesh generator has many options. I believe you can tune it to meet your needs. If not let me know.

**Reply by RECHOA** [reply]
If the mesh file created by Cubit has 3 merged blocks:
![pic](https://user-images.githubusercontent.com/33534979/191332515-e4446b96-cc0a-42f1-8e2b-9a3125358fe3.jpg)
the cohesive zone model (CZM) is applied in volume1 and volume2, i.e. the `BreakMeshByBlockGenerator `only apply in block1 and block2, not block3. How could this be realized?

**Reply by jiangwen84** [reply]
Use the 'block_pair' option. See https://mooseframework.inl.gov/source/meshgenerators/BreakMeshByBlockGenerator.html
